src/hotspot/cpu/aarch64/aarch64_vector.ad

```diff
@@ -2854,11 +2854,11 @@ instruct reduce_addL_sve(iRegLNoSp dst, iRegL isrc, vReg vsrc, vRegD tmp) %{
 %}
 
 // reduction addF
-// Floating-point addition is not associative, so the rules for AddReductionVF
-// on NEON can't be used to auto-vectorize floating-point reduce-add.
-// Currently, on NEON, AddReductionVF is only generated by Vector API.
 instruct reduce_add2F_neon(vRegF dst, vRegF fsrc, vReg vsrc) %{
-  predicate(UseSVE == 0 && Matcher::vector_length(n->in(2)) == 2);
+  // Non-strictly ordered floating-point add reduction for vector length of 64-bit. As an
+  // example, this rule can be reached from the VectorAPI (which allows for non-strictly ordered
+  // add reduction).
+  predicate(Matcher::vector_length(n->in(2)) == 2 && !n->as_Reduction()->requires_strict_order());
```

Comment:
Hi, thank you for your comments. I will make these changes in the next PS. Maybe it's a good idea to add "non_strict_order" as not everyone might check the ad file to read comments and for a quick glance in an opto assembly output for example, it might be helpful to understand what these instructions mean.

Reply:
Exactly!